components/script/dom/attr.rs

```diff
@@ -244,6 +247,24 @@ impl<'a> AttrHelpers<'a> for JSRef<'a, Attr> {
         &self.extended_deref().local_name
     }
 
+    fn set_owner(self, owner: Option<JSRef<Element>>) {
+        let ns = self.namespace.clone();
+        match (self.owner.get().root().r(), owner) {
+            (None, Some(new)) =>
+                // Already in the list of attributes of new owner.
+                assert!(new.get_attribute(ns, &self.local_name).root().r() == Some(self)),
```

Comment:
okay, leave it then. We probably should move some asserts to debug asserts later, but for now I guess we can stick to asserts everywhere